src/geojson.rs

```diff
@@ -27,25 +28,28 @@ pub fn write(objs: &BTreeMap<OsmId, OsmObj>, out: impl io::Write) -> Result<()>
 
 fn to_feature(obj: &OsmObj, all_objs: &BTreeMap<OsmId, OsmObj>) -> Option<geojson::GeoJson> {
     let tags = obj.tags();
-    let name = tags.get("name")?;
+    let name = [tags.get("name:prefix"), tags.get("name")]
+        .iter()
+        .filter_map(|&f| f)
+        .join(" ");
```

Comment:
Vorher hast du alles rausgeschmissen. Mit dieser Änderung ist `name` (und `name:prefix` auch) komplett optional, d.h., falls z.B. keines der beiden Felder gesetzt ist, hast du einen leeren String in `name`. Passt das zu dem, was du willst? Dichter an der vorherigen API wäre, `name` weiter required zu lassen und vielleicht einen prefix vorzuhängen, z.B.,

```rust
    let name = {
        let n = tags.get("name")?;
        tags.get("name:prefix")
            .map(|p| format!("{p} {n}"))
            .unwrap_or(n.to_string())
    };
```

Unabhänig davon frage ich mich, ob man wirklich `prefix` mit `name` vermengen sollte hier bei der Extraktion. Ich weiss, dass das daher kommt, dass `name` nicht eindeutig ist (z.B., `name="Krakow am See"` mit `prefix=Amt` und `prefix=Stadt`), aber am Ende scheint mir das ein consumer/display concern. Vielleicht einfach das alte required `name` field beibehalten und ein neues optionales Feld `name:prefix` emitten?

Reply:
> Mit dieser Änderung ist name (und name:prefix auch) komplett optional

Guter Punkt, vorher wäre der Algo ausgestiegen 🙄

> Unabhänig davon frage ich mich, ob man wirklich prefix mit name vermengen sollte

Ich stelle mir das im Frontend vor, wenn zweimal Krakow am See in der Suche erscheint und zwei unterschiedliche Entities dahinterstecken. Eine Differenzierung scheint mir angebracht.

Ich denke bei der Implementierung bereits an das MongoDB Datenmodell. Dort habe ich auf `properties.name` einen Suchindex laufen. Es macht aus meiner Sicht also schon Sinn, das bereits im Datenbestand zu mergen.